test/solvers/advanced/douglas_rachford_primal_dual_test.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test for the Douglas-Rachford solver."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import pytest
+import odl
+import odl.solvers as odls
+
+from odl.util.testutils import all_almost_equal
+
+
+def test_l1():
```

Comment:
Yeah I don't know. Hard to test these buggers. What you could do, though, is a matrix-vector case and use L2 for both functionals (we don't test the proximals). That you can compare to the Tikhonov least squares solution.